pipelinesScripts/validator/utils/utils.go

```diff
@@ -39,3 +52,69 @@ func getRootPath() (string, error) {
 	}
 	return absRootPath, nil
 }
+
+func GetPluginsDescriptor() (r []*PluginDescriptor, err error) {
+	rootPath, err := getRootPath()
+	if err != nil {
+		return nil, err
+	}
+	fileInfos, err := ioutil.ReadDir(filepath.Join(rootPath, PluginDescriptorDir))
+	if err != nil {
+		return nil, err
+	}
+	var results []*PluginDescriptor
+	for _, file := range fileInfos {
+		fDescriptor, err := ReadDescriptor(filepath.Join(PluginDescriptorDir, file.Name()))
+		if err != nil {
+			return nil, err
+		}
+		results = append(results, fDescriptor)
+	}
+	return results, nil
+}
+
+// Returns the plugins repository owner and name lowecase. e.g.: https://github.com/JFrog/jfrog-CLI-plugins => (jfrog, jfrog-cli-plugins)
+func ExtractRepoDetails(pluginRepository string) (owner string, repo string) {
+	pluginRepository = strings.Replace(pluginRepository, "https://github.com/", "", 1)
+	splitted := strings.Split(pluginRepository, "/")
+	return strings.ToLower(splitted[0]), strings.ToLower(splitted[1])
+}
+
+func UpdateGoDependency(runAt, DepName, depVersion string) (err error) {
+	// Keep the stdout clean. Ignoring 'go get' cmd output.
+	_, err = RunCommand(runAt, true, "go", "get", DepName+"@"+depVersion)
+	if err != nil {
+		fmt.Println("Go Get failed for at" + runAt)
+	}
+	return
+}
+
+// PluginDescriptor describes the plugin descriptor yml.
+type PluginDescriptor struct {
+	// Mandatory fields:
+	PluginName  string   `yaml:"pluginName"`  // Example: RT-FS
+	Version     string   `yaml:"version"`     // Example: 1.0.0
+	Repository  string   `yaml:"repository"`  // Example: https://github.com/jfrog/jfrog-cli-plugins-reg.git
+	Maintainers []string `yaml:"maintainers"` // Example: ['frog1', 'frog2']
+
+	// Optional fields:
+	RelativePath string `yaml:"relativePath"` // Example: rt-fs
+	Branch       string `yaml:"branch"`       // Example: rel-1.0.0
+	Tag          string `yaml:"tag"`          // Example: 1.0.0
+}
+
+func ReadDescriptor(filePath string) (*PluginDescriptor, error) {
```

Comment:
Public function when should have been private.
Notice that there may be more instances, which haven't pointed out.

Reply:
Can't make this private since `validateDescriptor` uses it.